paychmgr/paych.go

```diff
@@ -279,16 +232,7 @@ func (pm *Manager) AddVoucher(ctx context.Context, ch address.Address, sv *paych
 		return types.NewInt(0), err
 	}
 
-	laneState, err := pm.laneState(ctx, ch, uint64(sv.Lane))
-	if err != nil {
-		return types.NewInt(0), err
-	}
-
-	if minDelta.GreaterThan(types.NewInt(0)) && laneState.Nonce > sv.Nonce {
-		return types.NewInt(0), xerrors.Errorf("already storing voucher with higher nonce; %d > %d", laneState.Nonce, sv.Nonce)
-	}
-
-	// look for duplicates
+	// Check if the voucher has already been added
```

Comment:
When there was a duplicate voucher, the existing code would store a copy of the voucher with each different proof passed to AddVoucher(), but this caused problems in other parts of the code, so I removed that behaviour.
Now there will only be one copy of the voucher, and any new proof overwrites the existing proof.